mk/cache-test.mk

```diff
@@ -0,0 +1,49 @@
+CACHE_TEST_DIR := tests/cache
+CACHE_BUILD_DIR := build/cache
+TARGET := test_cache.out
```

Comment:
The naming scheme violates the consistency.
* `.out` is the suffix for the files generated by test programs. Typically, the text output.
* For host executables, there is no specific suffix.

Here, simply name it `test_cache`.